linode/manager
# feat: [M3-6989] - Improve helper text accessibility

## Description 📝
Highlight the Pull Request's context and intentions.

## Changes  🔄
List any change relevant to the reviewer.
- Changes the HTML element that is rendered by the `Typography` component from `<p>` to `<span>`. This is done using the `component` prop.
- Sets an `id` attribute in the `Typography` component to associate the helper text to the `<input />` element.

## Target release date 🗓️
10/14/2024

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-10-03 at 9 44 46 AM](https://github.com/user-attachments/assets/3da2f5dd-2b2e-4482-8d70-28111dbac0df) | ![Screenshot 2024-10-03 at 9 41 41 AM](https://github.com/user-attachments/assets/3b5bf0db-eec4-492f-b708-e13301b051d7) |
| ![Screenshot 2024-10-03 at 9 46 03 AM](https://github.com/user-attachments/assets/7c519453-0606-4b3b-9272-3048c849db62) | ![Screenshot 2024-10-03 at 9 41 56 AM](https://github.com/user-attachments/assets/939ed282-7b46-4a47-a519-6815cb4dbfe9) |


## How to test 🧪

### Prerequisites
(How to setup test environment)
- Visit the `linodes/create` page.
- Turn on Apple's VoiceOver utility.
- Make sure your volume is up.

### Reproduction steps
(How to reproduce the issue, if applicable)
- Traverse the form fields using tab navigation on your keyboard.
- Pay close attention to the output read by the VoiceOver utility when the following form fields have been focused.
  - `Region`
  - `Linode Plan`
  - `Disk Encryption`
  - `VPC`
  - `Firewall` 
  - `Backups`
  - `Private IP`

### Verification steps
(How to verify changes)
- Verify that upon focusing on the interactive fields, the helper text is read by the VoiceOver utility. 
- Verify that the form field labels and values are also read by the VoiceOver utility.
- Verify that there are no visual regressions.

## As an Author I have considered 🤔

*Check all that apply*
